smarts/core/argoverse_map.py

```diff
@@ -975,36 +972,36 @@ def waypoint_paths(
         # No route provided, so generate paths based on the closest lanepoints
         waypoint_paths = []
         closest_lps: List[LanePoint] = self._lanepoints.closest_lanepoints(
-            pose, within_radius=within_radius
+            pose, within_radius=within_radius, maximum_count=15
         )
         closest_lane: RoadMap.Lane = closest_lps[0].lane
 
         # First, see if we are in a junction and need to do something special
         if closest_lane.in_junction:
-            junction_lanes: Set[RoadMap.Lane] = set([closest_lane])
+            # Get the set of all nearby junction lanes with similar heading
+            junction_lanes: Set[RoadMap.Lane] = set()
             for lp in closest_lps:
                 rel_heading = lp.pose.heading.relative_to(pose.heading)
-                if (
-                    lp.lane.in_junction
-                    and abs(rel_heading) < np.pi / 2
-                    and lp.lane.contains_point(pose.point)
-                ):
+                if lp.lane.in_junction and abs(rel_heading) < np.pi / 2:
                     junction_lanes.add(lp.lane)
 
-            paths = set()
+            # Get set of all lanes leading through the junction
+            wp_lanes = set()
             for junction_lane in junction_lanes:
-                paths_for_lane = self._resolve_in_junction(junction_lane)
-                for path in paths_for_lane:
-                    paths.add(tuple(path))
-
-            for road_ids in paths:
-                new_paths = self._waypoint_paths_along_route(
-                    pose.point, lookahead, road_ids
```

Comment:
If I remember correctly, this was done to constrain the waypoints to route for cases where there is a mission. Does that still work after this change?

Reply:
That happens above in a different place. It probably never should have been used here in the first place.